README.rst

```diff
@@ -255,97 +255,14 @@ Other non-official Python wrappers for GMT (not maintained):
 Compatibility with GMT/Python/NumPy versions
 --------------------------------------------
```

Comment:
Although we're not adopting [SPEC 0](https://scientific-python.org/specs/spec-0000/) yet in this PR, I feel the section heading used in SPEC 0 makes more sense:

```suggestion
Minimum Supported GMT/Python/NumP Versions
------------------------------------------
```